internal/provider/resource_virtual_machine_group_test.go

```diff
@@ -0,0 +1,137 @@
+	"context"
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/jimeh/undent"
+)
+
+func init() { //nolint:gochecknoinits
+	resource.AddTestSweepers("katapult_virtual_machine_group",
+		&resource.Sweeper{
+			Name: "katapult_virtual_machine_group",
+			F:    testSweepVMGroups,
+		})
+}
+
+func testSweepVMGroups(_ string) error {
+	m := sweepMeta()
+	ctx := context.TODO()
+
+	vmgs, _, err := m.Client.VirtualMachineGroups.List(
+		ctx, m.OrganizationRef(),
+	)
+	if err != nil {
+		return err
+	}
+
+	for _, vmg := range vmgs {
+		log.Printf(
+			"[DEBUG]  - Deleting Virtual Machine Group %s (%s)\n",
+			vmg.ID, vmg.Name,
+		)
+		_, err := m.Client.VirtualMachineGroups.Delete(ctx, vmg)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func TestAccKatapultVMGroup_basic(t *testing.T) {
+	tt := newTestTools(t)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckKatapultVMGroupDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `resource "katapult_virtual_machine_group" "web" {
+					name = "test group"
+				}`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"katapult_virtual_machine_group.web",
+						"name",
+						"test group",
+					),
+				),
+			},
+			{
+				ResourceName:      "katapult_virtual_machine_group.web",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccKatapultVMGroup_segregate(t *testing.T) {
+	tt := newTestTools(t)
+
+	name := tt.ResourceName("web-segregate")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckKatapultVMGroupDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: undent.Stringf(`
+					resource "katapult_virtual_machine_group" "segregate" {
+					  segregate = true
+					  name = "%s"
+					}`,
+					name,
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"katapult_virtual_machine_group.segregate",
+						"segregate", "true",
+					),
+					resource.TestCheckResourceAttr(
+						"katapult_virtual_machine_group.segregate",
+						"name", name,
+					),
+				),
+			},
+			{
+				ResourceName:      "katapult_virtual_machine_group.segregate",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckKatapultVMGroupDestroy(
+	tt *testTools,
+) resource.TestCheckFunc {
+	m := tt.Meta
+
+	return func(s *terraform.State) error {
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "katapult_virtual_machine_group" {
+				continue
+			}
+
+			vmg, _, err := m.Client.VirtualMachineGroups.GetByID(
+				tt.Ctx, rs.Primary.ID)
```

Comment:
This is just a personal annoyance, but I feel the code looks more balanced and is easier to skim without reading if the `)` is on it's own line in this kind of situation.
```suggestion
				tt.Ctx, rs.Primary.ID,
			)
```